build/0001-cloudflare.patch

```diff
@@ -0,0 +1,58 @@
+diff --git a/dh/x25519/BUILD.bazel b/dh/x25519/BUILD.bazel
+index d569bc1..13a942d 100644
+--- a/dh/x25519/BUILD.bazel
++++ b/dh/x25519/BUILD.bazel
+@@ -12,7 +12,9 @@ go_library(
+         "doc.go",
+         "key.go",
+         "table.go",
++        "//math/fp25519:fp_amd64.h",
+     ],
++    cgo = True,
```

Comment:
The diff here is quite large - does this reenable CGO_ENABLED writ large? Can we avoid that?

Reply:
> CGO_ENABLED writ large

Not sure what you mean with this?

This enables CGO for this library.  I don't know Go tooling enough to know whether that transitively enables it in the dependent libraries though.

Reply:
Just trying to page in why this was needed in the first place, it's just that gazelle (which also generates build files for external dependencies) just doesn't deal with the CGO structure of this particular dependency properly.  The library needs to be build with CGO either way, so that shouldn't change anything here I believe.

Reply:
I don't quite understand this - this library requires CGO but the rest of the binary build won't?

Maybe I don't understand how that could work with Go.

I do think we should not regress and switch to CGO_ENABLED builds, it makes cross compilation much more challenging and as Go developers say, CGO is not go. There are downsides to CGO_ENABLED=1, and we finally extricated ourselves from that a year or so ago. Going back will lead to surprises.

https://dave.cheney.net/2016/01/18/cgo-is-not-go

Reply:
Actually turns out the `cgo = True` was not actually needed, and I just copy-pasta'd too much stuff from the internet :sweat_smile: 

I do think this needs a little bit of additional patching to make it work on non-linux platforms, but that should be fairly straightforward.